src/connection/connection_support.c

```diff
@@ -2945,3 +2957,213 @@ css_conn_entry::init_pending_request ()
+       */
+      if (css_Server_inhibit_connection_socket)
+	{
+	  server_port_id = -1;
+	}
+      else
+	{
+	  server_port_id = css_open_server_connection_socket ();
+	}
+      response = htonl (server_port_id);
+      css_net_send (conn, (char *) &response, sizeof (int), -1);
+      /* this connection remains our only contact with the master */
+      return conn;
+
+    case SERVER_REQUEST_ACCEPTED:
+#if defined(WINDOWS)
+      /* Windows can't handle this style of connection at all */
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ERR_CSS_ERROR_DURING_SERVER_CONNECT, 1, message_to_master);
+      goto fail_end;
+#else /* WINDOWS */
+      /* send the "pathname" for the datagram */
+      /* be sure to open the datagram first.  */
+      pname = std::filesystem::temp_directory_path ();
+
+#if !defined (SERVER_MODE)
+      pname += "/csql_tcp_setup_server" + std::to_string (getpid ());
+#else // SERVER_MODE
+      pname += "/cubrid_tcp_setup_server" + std::to_string (getpid ());
+#endif // SERVER_MODE
+      (void) unlink (pname.c_str ());	// make sure file is deleted
+      if (!css_tcp_setup_server_datagram (pname.c_str (), &socket_fd))
+	{
+	  (void) unlink (pname.c_str ());
+	  er_set_with_oserror (ER_ERROR_SEVERITY, ARG_FILE_LINE, ERR_CSS_ERROR_DURING_SERVER_CONNECT, 1,
+			       message_to_master);
+	  goto fail_end;
+	}
+      if (css_send_data (conn, rid, pname.c_str (), pname.length () + 1) != NO_ERRORS)
+	{
+	  (void) unlink (pname.c_str ());
+	  close (socket_fd);
+	  er_set_with_oserror (ER_ERROR_SEVERITY, ARG_FILE_LINE, ERR_CSS_ERROR_DURING_SERVER_CONNECT, 1,
+			       message_to_master);
+	  goto fail_end;
+	}
+      if (!css_tcp_listen_server_datagram (socket_fd, &datagram_fd))
+	{
+	  (void) unlink (pname.c_str ());
+	  css_free_conn (conn);
+	  close (socket_fd);
+	  er_set_with_oserror (ER_ERROR_SEVERITY, ARG_FILE_LINE, ERR_CSS_ERROR_DURING_SERVER_CONNECT, 1,
+			       message_to_master);
+	  return NULL;
+	}
+      // success
+      (void) unlink (pname.c_str ());
+      css_free_conn (conn);
+      close (socket_fd);
+      return (css_make_conn (datagram_fd));
+#endif /* WINDOWS */
+    }
+
+fail_end:
+  css_free_conn (conn);
+  return NULL;
+}
+
+CSS_CONN_ENTRY *
+css_common_connect (CSS_CONN_ENTRY * conn, unsigned short *rid, const char *host_name,
+		    int connect_type, const char *message, int message_length, int port, int timeout, bool send_magic)
+{
+  SOCKET fd;
+
+#if !defined (SERVER_MODE)
+#if !defined (WINDOWS)
+  if (timeout > 0)
+    {
+      /* timeout in milli-seconds in css_tcp_client_open_with_timeout() */
+      fd = css_tcp_client_open_with_timeout (host_name, port, timeout * 1000);
+    }
+  else
+    {
+      fd = css_tcp_client_open_with_retry (host_name, port, true);
+    }
+#else /* !WINDOWS */
+  fd = css_tcp_client_open_with_retry (host_name, port, true);
+#endif /* WINDOWS */
+
+#else // SERVER_MODE
+  fd = css_tcp_client_open ((char *) host_name, port);
+#endif // SERVER_MODE
```

Comment:
The next code is equivalent, but fewer lines:
```suggestion
#if !defined (WINDOWS) && !defined (SERVER_MODE)
  if (timeout > 0)
    {
      /* timeout in milli-seconds in css_tcp_client_open_with_timeout() */
      fd = css_tcp_client_open_with_timeout (host_name, port, timeout * 1000);
    }
  else
#else /* WINDOWS || SERVER_MODE */
    {
      fd = css_tcp_client_open (host_name, port);
    }
#endif /* WINDOWS || SERVER_MODE */
```